pa11ycrawler/spiders/edx.py

```diff
@@ -84,6 +129,10 @@ def parse_item(self, response):
         @returns requests 0 0
         @scrapes url request_headers accessed_at page_title
         """
+        # if we got redirected to a login page, then login
+        if URLObject(response.url).path in ("/login", "/register"):
+            yield self.make_login_request(response)
```

Comment:
Because parse callbacks (including this one) are supposed to be generators, which means they can `yield` multiple times. In this case, we can actually `yield` twice -- once with a `scrapy.Request` instance that will handle logging back in, and once with an `A11yItem` instance, to pass to the pipeline (including the `Pa11yPipeline` that runs `pa11y` on the login page). Does that make sense?